qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1297,23 +1301,50 @@ def substitute_node(self, node, op, inplace=False):
                 )
             )
 
+        # This might include wires that are inherent to the node, like in its `condition` or
+        # `target` fields, so might be wider than `node.op.num_{qu,cl}bits`.
+        current_wires = {wire for _, _, wire in self.edges(node)}
+        new_wires = set(node.qargs) | set(node.cargs)
+        if (new_condition := getattr(op, "condition", None)) is not None:
+            new_wires.update(condition_resources(new_condition).clbits)
```

Comment:
Is there a mapping or check that needs to be done here to ensure `op.condition` exists in the dag already?

Reply:
If you mean the wires, it happens lower down, I think: https://github.com/Qiskit/qiskit-terra/blob/60cd949fc41915c1a12fb8faa3e4c1ea3e6f94b5/qiskit/dagcircuit/dagcircuit.py#L1331-L1337.

(Looks like there might be a comment that's out of date there though haha!)

Reply:
Ah, yeah that'll do it. It just slipped my mental context that this would validate wires.